software_scale_context_test.go

```diff
@@ -0,0 +1,104 @@
+package astiav_test
+
+import (
+	"image"
+	"testing"
+
+	"github.com/asticode/go-astiav"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSoftwareScaleContext(t *testing.T) {
+	f1 := astiav.AllocFrame()
+	require.NotNil(t, f1)
+	defer f1.Free()
+
+	f2 := astiav.AllocFrame()
+	require.NotNil(t, f2)
+	defer f2.Free()
+
+	f3 := astiav.AllocFrame()
+	require.NotNil(t, f3)
+	defer f3.Free()
+
+	srcW := 100
+	srcH := 100
+	srcPixelFormat := astiav.PixelFormatYuv420P
+	dstW := 200
+	dstH := 200
+	dstPixelFormat := astiav.PixelFormatRgba
+
+	f1.SetHeight(srcH)
+	f1.SetWidth(srcW)
+	f1.SetPixelFormat(srcPixelFormat)
+	require.NoError(t, f1.AllocBuffer(1))
+
+	swscf_1 := astiav.NewSoftwareScaleContextFlags(astiav.SoftwareScaleContextBilinear)
+	swsc := astiav.NewSoftwareScaleContext(srcW, srcH, srcPixelFormat, dstW, dstH, dstPixelFormat, swscf_1)
+	require.NotNil(t, swsc)
+	require.Equal(t, swsc.Flags(), swscf_1)
+
+	swscf_2 := astiav.NewSoftwareScaleContextFlags(astiav.SoftwareScaleContextPoint)
+	swsc.SetFlags(swscf_2)
+	require.Equal(t, swsc.Flags(), swscf_2)
+
+	require.NoError(t, swsc.PrepareDestinationFrameForScaling(f2))
+	require.Equal(t, dstH, swsc.ScaleFrame(f1, f2))
+
+	require.Equal(t, dstW, f2.Height())
+	require.Equal(t, dstH, f2.Width())
+	require.Equal(t, dstPixelFormat, f2.PixelFormat())
+
+	i1, err := f2.Data().Image()
```

Comment:
I don't feel this "paragraph" is testing anything useful since it's testing whether `FrameData` works properly, not whether scaling worked properly